spot_driver/launch/spot_driver.launch.py

```diff
@@ -104,9 +105,43 @@ def create_point_cloud_nodelets(
     return composable_node_descriptions
 
 
-def launch_setup(context: LaunchContext, ld: LaunchDescription) -> None:
+def spot_has_arm(context: LaunchContext) -> bool:
+    # Check if spot has an arm by logging in and instantiating a SpotWrapper
+    config_file_path = LaunchConfiguration("config_file").perform(context)
+    spot_name = LaunchConfiguration("spot_name").perform(context)
     logger = logging.getLogger("spot_driver_launch")
+    username = os.getenv("BOSDYN_CLIENT_USERNAME")
+    password = os.getenv("BOSDYN_CLIENT_PASSWORD")
+    hostname = os.getenv("SPOT_IP")
+    port = int(os.getenv("SPOT_PORT", "0"))  # TODO should the user be able to specify a port via config file?
+    # parse the yaml to determine if login information is set there
+    if os.path.isfile(config_file_path):
+        with open(config_file_path, "r") as config_yaml:
+            try:
+                config_dict = yaml.safe_load(config_yaml)
+                if ("/**" in config_dict) and ("ros__parameters" in config_dict["/**"]):
+                    ros_params = config_dict["/**"]["ros__parameters"]
+                    # only set username/password/hostname if they were not already set as environment variables.
+                    if (username is None) and ("username" in ros_params):
```

Comment:
nit: It's a long shot, but maybe use `if (not username) and (...):` to protect against the case when it is an empty string?